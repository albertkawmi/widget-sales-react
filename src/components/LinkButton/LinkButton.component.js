import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import './LinkButton.css'

const LinkButton = ({
  to,
  text = '',
  linkClassName = '',
  buttonClassName = '',
  options = {}
}) => (
  <Link className={linkClassName} to={to}>
    <button className={buttonClassName} disabled={options.disabled}>
      {text}
    </button>
  </Link>
)

LinkButton.propTypes = {
  to: PropTypes.string.isRequired,
  text: PropTypes.string,
  linkClassName: PropTypes.string,
  buttonClassName: PropTypes.string,
  options: PropTypes.object
}

LinkButton.defaultProps = {
  text: '',
  linkClassName: '',
  buttonClassName: '',
  options: {
    disabled: false
  }
}

export default LinkButton
