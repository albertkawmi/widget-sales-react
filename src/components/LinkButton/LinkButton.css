button {
  background: white;
  border: 0;
  border: solid 1px silver;
  border-bottom: solid 2px #2bafba;
  border-radius: 4px;
  cursor: pointer;
}

button:hover {
  background: lemonchiffon;
}

button:disabled {
  background: white;
  border-bottom: solid 2px silver;
  cursor: default;
}
