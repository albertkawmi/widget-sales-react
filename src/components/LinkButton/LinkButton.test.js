/* eslint-env jest */
import React from 'react'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import LinkButton from './LinkButton.component'

Enzyme.configure({ adapter: new Adapter() })

describe('<LinkButton />', () => {
  const button = Enzyme.shallow(
    <LinkButton to="/amazing/page" />
  )

  it('renders a help message for mobile users', () => {
    expect(button).toMatchSnapshot()
  })
})
