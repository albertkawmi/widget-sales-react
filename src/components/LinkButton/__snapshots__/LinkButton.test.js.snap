// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkButton /> renders a help message for mobile users 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LinkButton
    buttonClassName=""
    linkClassName=""
    options={
        Object {
            "disabled": false,
          }
    }
    text=""
    to="/amazing/page"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <button
        className=""
        disabled={false}
/>,
      "className": "",
      "replace": false,
      "to": "/amazing/page",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "",
        "className": "",
        "disabled": false,
      },
      "ref": null,
      "rendered": "",
      "type": "button",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <button
          className=""
          disabled={false}
/>,
        "className": "",
        "replace": false,
        "to": "/amazing/page",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "",
          "disabled": false,
        },
        "ref": null,
        "rendered": "",
        "type": "button",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
